include/storage/flash_map.h

```diff
@@ -262,15 +267,63 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
 #define FLASH_AREA_LABEL_STR(lbl) \
 	DT_PROP(DT_NODE_BY_FIXED_PARTITION_LABEL(lbl), label)
 
-#define FLASH_AREA_ID(label) \
-	DT_FIXED_PARTITION_ID(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
+/**
+ * Obtain pointer to flash_area objet directly from DTS label of fixed partition.
+ *
+ * @param[in] label Fixed partition label to use; this C identifier type label, which menas that
+ *	      instead of for example image-0, image_0 sh uld be provided.
+
+ * @return const struct flash_area * to flash_area object or NULL.
+ */
+#define FLASH_AREA(label)								\
+	(&_CONCAT(_flash_map_area_,							\
```

Comment:
Does this really return NULL on missing partitions? It looks like it would cause a linker error instead.

Reply:
> 
> 
> Does this really return NULL on missing partitions? It looks like it would cause a linker error instead.

Sorry, the comment was from previous version where it did, but it is better to let it fail at compilation/linking.